### DISCUSSION TITLE
azure_servicebus_scaler	error {"error": "no azure identity found for request clientID \n"}

### DISCUSSION THREAD

**Discussion by scubakiz** [kedacore/keda#2180] [Q&A / Need Help]
I'm using Azure Identity with Pod Identity and everything works except for KEDA.  The KEDA operator keeps returning this error during the polling interval.

```bash
2021-10-10T19:57:07.064Z	ERROR	azure_servicebus_scaler	error	{"error": "no azure identity found for request clientID \n"}
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).isScaledObjectActive
	/workspace/pkg/scaling/scale_handler.go:228
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).checkScalers
	/workspace/pkg/scaling/scale_handler.go:211
github.com/kedacore/keda/v2/pkg/scaling.(*scaleHandler).startScaleLoop
	/workspace/pkg/scaling/scale_handler.go:145
```

I added this label to everything (the pod, deployment, scaledobject, Keda operator):  
```bash
  labels:
    aadpodidbinding: jet-identity
```

I updated this setting:

```bash
helm upgrade keda kedacore/keda --namespace keda --set podIdentity.activeDirectory.identity=jet-identity
```

And confirmed that the ScaledObject and HPA have the correct label.

If turn off KEDA and run the deployment normally, it works correctly.  A Pod is created (containing an Azure Functions trigger).  The trigger fires and processes the messages in the Service Bus Queue using the Pod Identity.

If I set the idleReplicaCount/minReplicaCount =1 in a SO, the SO and HPA are created and have a status of AbleToScale.  Again 1 pod is created and works correctly. 

However, the HPA metric looks like this because the scaler is unable to read the number of messages in the Service Bus:

```bash
azure-servicebus-jetbus-servicebus-windows-net-jetsbqueue on unknown / 1
```

If I set the idleReplicaCount/minReplicaCount = 0, no Pods are ever created.

NOTE:  I have another deployment in this namespace using Azure Functions runtime to trigger on an HTTP request and write messages to this same ServiceBus Queue.  It uses the same connection string and works fine using Pod Identity to write to the queue.

This is the connection string to the queue:

```bash
 - name: SERVICEBUS_CONNECTIONSTRING
   value: Endpoint=sb://jetbus.servicebus.windows.net/;Authentication=ManagedIdentity;EntityPath=jetsbqueue    
```

**Comment by jeffhollan** [comment]
Can you confirm in your `ScaledObject` is referencing a `TriggerAuthentication` object that tells KEDA to use pod identity to authenticate?

There's a sample in our docs here https://keda.sh/docs/2.4/scalers/azure-service-bus/
